src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,68 @@
+import classNames from 'classnames';
+import { useEffect, useState } from 'react';
+import { useAppDispatch } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { actions as todoActions } from '../../features/currentTodo';
+
+type Props = {
+  todo: Todo
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => {
+  const [buttonHandler, setButtonHandler] = useState(false);
+  const dispatch = useAppDispatch();
+
+  const todoButtonHandler = () => {
+    setButtonHandler(!buttonHandler);
+  };
+
+  const selectedTodoSaver = () => {
+    if (buttonHandler) {
+      dispatch(todoActions.setTodo(todo));
+    }
+
+    dispatch(todoActions.removeTodo());
+  };
+
+  useEffect(() => {
+    selectedTodoSaver();
+  }, [buttonHandler]);
```

Comment:
Same here. You don't use buttonHandler here, so add the `selectedTodoSaver` to the dependency array and wrap this `selectedTodoSaver` function into useCallback() with `buttonHandler` in the array